src/sentry/replays/usecases/query/__init__.py

```diff
@@ -440,6 +441,69 @@ def make_full_aggregation_query(
     )
 
 
+def make_full_aggregation_query_with_short_id(
+    fields: list[str],
+    replay_id_prefix: str,
+    project_ids: list[int],
+    period_start: datetime,
+    period_end: datetime,
+    request_user_id: int | None,
+    limit: int,
+) -> Query:
+    """Return a query to fetch a replay with a short ID - an 8-character replay ID prefix.
+    This query does not make use of the replay_id index and can potentially scan all rows in the time range and project list.
+
+    Arguments:
+        fields -- if non-empty, used to query a subset of fields. Corresponds to the keys in QUERY_ALIAS_COLUMN_MAP.
+    """
+
+    if len(replay_id_prefix) != 8:
+        raise ValueError("Invalid short ID. Must be 8 hexadecimal characters.")
+
+    try:
+        int(replay_id_prefix, 16)
+    except ValueError as e:
+        raise ValueError("Invalid short ID. Must be 8 hexadecimal characters.") from e
+
+    replay_id_prefix = replay_id_prefix.lower()
+
+    from sentry.replays.query import select_from_fields
+
+    select = select_from_fields(fields, user_id=request_user_id)
+
+    return Query(
```

Comment:
could we potentially do a pre-query to only look at the replay id? but no worries if you want to leave it as is